promql/engine.go

```diff
@@ -1023,6 +1026,7 @@ func checkAndExpandSeriesSet(ctx context.Context, expr parser.Expr) (annotations
 			}
 		}
 		e.Series = series
+		ev.selectHints = e.SelectHints
```

Comment:
This sets whatever VectorSelector was encountered last in the AST walk; I find it unlikely this will be correct in a complex expression.

Reply:
Thanks, I'm thinking about this some more.

Reply:
Could you think of an expression that will be incorrect?

Reply:
I've pushed a [revision](https://github.com/prometheus/prometheus/pull/14495/commits/0f2520948de01bc740766971ef1f1acf12e1f504), that instead of picking `SelectHints` from the latest `VectorSelector` to execute, explicitly derives the start/end/step parameters for selecting info series. It does find the topmost `VectorSelector` in the first argument to the `info` call, in order to deduce the right query timestamp and/or offset.

I added three test cases for respectively lookback, `@` operator and `offset` operator.

WDYT? @MichaHoffmann already had a look, and thinks it makes sense.